### PR TITLE
improve color-support-detection

### PR DESCRIPTION
**Pull request by mimmi20**
Because the current color-support detection does not really work with the PowerShell, I try to replace this with PHPUnits implementation.

I also try to add some checks to the startup script which I copied from PHPUnit.

### CODE REVIEW

File: bin/phing.php [hunk 1]
```diff
@@ -6,42 +6,58 @@
...
-// ---------------------------
+    die(1);
+}
 
-/* set include paths */
-set_include_path(
-            __DIR__ . '/../classes' .
-            PATH_SEPARATOR .
-            get_include_path()
-        );
```
Comment (siad007) [inline bin/phing.php]: Removing this lines could break the code in Phing.php line 1567
```
        if (!defined('PHP_CLASSPATH')) {
            define('PHP_CLASSPATH', get_include_path());
        }

        self::setProperty('php.classpath', PHP_CLASSPATH);
```

File: bin/phing.php [hunk 2]
```diff
@@ -6,42 +6,58 @@
...
+if (version_compare('7.1.0', PHP_VERSION, '>')) {
+    fwrite(
+        STDERR,
+        sprintf(
+            'This version of Phing is supported on PHP 7.1, PHP 7.2, PHP 7.3 and PHP 7.4.' . PHP_EOL .
+            'You are using PHP %s (%s).' . PHP_EOL,
+            PHP_VERSION,
+            PHP_BINARY
+        )
+    );
```
Comment (siad007) [inline bin/phing.php]: color support?

**Comment by siad007** [comment]
@mimmi20 could you post the version of your windows and powershell? I have colored output on win 10 (powershell and cygwin)

**Comment by mimmi20** [comment]
> could you post the version of your windows and powershell?

Windows: 10.0.17763.973
PowerShell: 5.1.17763.771

File: bin/phing.php [hunk 3]
```diff
@@ -6,42 +6,58 @@
...
-/**
-* Code from Symfony/Component/Console/Output/StreamOutput.php
-*/
-function hasColorSupport()
-{
-    if (DIRECTORY_SEPARATOR == '\\') {
-        return 0 >= version_compare('10.0.10586', PHP_WINDOWS_VERSION_MAJOR.'.'.PHP_WINDOWS_VERSION_MINOR.'.'.PHP_WINDOWS_VERSION_BUILD)
-        || false !== getenv('ANSICON')
-        || 'ON' === getenv('ConEmuANSI')
-        || 'xterm' === getenv('TERM');
```
[the remark this answers is not in the document]
Comment (mimmi20) [reply bin/phing.php]: I dont like the idea of copying some code in fear of more dependencies. If you copy the code, you have to maintain it.

File: bin/phing.php (hunk 1 above)
Comment (mimmi20) [reply bin/phing.php]: The classes directory is added to the include path via composer.json

**Comment by mimmi20** [comment]
@siad007 I splitted the PR.

File: bin/phing.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (siad007) [reply bin/phing.php]: > I dont like the idea of copying some code in fear of more dependencies. If you copy the code, you have to maintain it.

But if you take dependencies you might trap into some pitfalls see #1223 #1224 #1160 
nevertheless the code has to be fixed here!

**Pull request closed without merging**